graphql/resolver.go

```diff
@@ -68,5 +70,32 @@ func New(apiURL string) Config {
 		}
 		return nil, ResourceNotFound.Send(ctx, "Could not find project")
 	}
+	c.Directives.RestrictProjectAccess = func(ctx context.Context, obj interface{}, next graphql.Resolver) (res interface{}, err error) {
+		user := gimlet.GetUser(ctx)
+		if user == nil {
+			return nil, Forbidden.Send(ctx, "user not logged in")
+		}
```

Comment:
You should use `mustHaveUser` here it handles panicking and logging an error as well. Since a missing user is pretty serious since it means they somehow got past auth.
https://github.com/evergreen-ci/evergreen/blob/d1f28a88613c3789e22ad4b1ddd5f2c3935b284c/graphql/util.go#L572